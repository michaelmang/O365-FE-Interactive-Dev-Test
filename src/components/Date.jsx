import classnames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';

import styles from './Date.module.scss';

const Date = ({ onDate, value }) => (
  <div className={classnames('field', styles.field)}>
    <div className="control">
      <input className="input" onChange={onDate} type="date" value={value} />
    </div>
  </div>
);

Date.propTypes = {
  onDate: PropTypes.func.isRequired,
  value: PropTypes.string.isRequired,
};

export default Date;
