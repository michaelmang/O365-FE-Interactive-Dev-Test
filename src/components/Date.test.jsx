import { shallow } from 'enzyme';
import React from 'react';

import Date from './Date.jsx';
import renderSnapshot  from '../testing/renderSnapshot.js';

jest.mock('./Date.module.scss', () => ({
  field: 'styles.field',
}));

describe('Date', () => {
  it('matches the saved snapshot', () => {
    let { root: actual } = extractComponents(renderComponent());

    expect(renderSnapshot(actual)).toMatchSnapshot();
  });

  describe('input', () => {
    it('passes down onChange prop with value of onDate', () => {
      let onDate = jest.fn();
      let { input: subject } = extractComponents(renderComponent({
        onDate,
      }));

      expect(subject).toHaveProp(
        'onChange',
        onDate,
      );
    });
  });

  const extractComponents = (root) => ({
    input: root.find('input'),
    root,
  });

  const renderComponent = (props) => (
    shallow(
      <Date
        onDate={() => {}}
        value="some date"
        {...props}
      />
    )
  );
});
