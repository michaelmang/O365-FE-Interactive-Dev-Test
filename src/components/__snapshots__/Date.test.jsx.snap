// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Date matches the saved snapshot 1`] = `
<div
  className="field styles.field"
>
  <div
    className="control"
  >
    <input
      className="input"
      onChange={[Function]}
      type="date"
      value="some date"
    />
  </div>
</div>
`;
